alphastats/gui/pages/05_LLM.py

```diff
@@ -17,171 +14,64 @@
-    st.session_state[StateKeys.GPT_SUBMITTED_CLICKED] = 0
-    st.session_state[StateKeys.GPT_SUBMITTED_COUNTER] = 0
+llm_config()
 
+st.markdown("#### Analysis")
 
-st.markdown("#### Configure LLM")
 
-c1, _ = st.columns((1, 2))
-with c1:
-    st.session_state[StateKeys.API_TYPE] = st.selectbox(
-        "Select LLM",
-        ["gpt4o", "llama3.1 70b"],
-        index=0 if st.session_state[StateKeys.API_TYPE] == "gpt4o" else 1,
-    )
+if StateKeys.LLM_INPUT not in st.session_state:
+    st.info("Create a Volcano plot first using the 'Analysis' page.")
+    st.stop()
 
-    if st.session_state[StateKeys.API_TYPE] == "gpt4o":
-        api_key = st.text_input("Enter OpenAI API Key", type="password")
-        set_api_key(api_key)
+volcano_plot, parameter_dict = st.session_state[StateKeys.LLM_INPUT]
 
-st.markdown("#### Analysis")
 c1, c2 = st.columns((1, 2))
 
 with c1:
-    method = select_analysis()
-    chosen_parameter_dict = helper_compare_two_groups()
-
-    method = st.selectbox(
-        "Differential Analysis using:",
-        options=["ttest", "anova", "wald", "sam", "paired-ttest", "welch-ttest"],
-    )
-    chosen_parameter_dict.update({"method": method})
-
-    # TODO streamlit doesnt allow nested columns check for updates
-
-    labels = st.checkbox("Add label", value=True)
-
-    draw_line = st.checkbox("Draw line", value=True)
-
-    alpha = st.number_input(
-        label="alpha", min_value=0.001, max_value=0.050, value=0.050
-    )
-
-    organism = st.number_input(
-        label="UniProt organism ID, for example human is 9606, R. norvegicus is 10116",
-        value=9606,
-    )
-    st.session_state[StateKeys.ORGANISM] = organism
-
-    min_fc = st.select_slider("Foldchange cutoff", range(0, 3), value=1)
-
-    plotting_parameter_dict = {
-        "labels": labels,
-        "draw_line": draw_line,
-        "alpha": alpha,
-        "min_fc": min_fc,
-    }
-
-    if method == "sam":
-        perm = st.number_input(
-            label="Number of Permutations", min_value=1, max_value=1000, value=10
-        )
-        fdr = st.number_input(
-            label="FDR cut off", min_value=0.005, max_value=0.1, value=0.050
-        )
-        chosen_parameter_dict.update({"perm": perm, "fdr": fdr})
-
-    plot_submitted = st.button("Plot")
-    if plot_submitted:
-        st.session_state[StateKeys.PLOT_SUBMITTED_CLICKED] += 1
-
-
-if (
-    st.session_state[StateKeys.PLOT_SUBMITTED_COUNTER]
-    < st.session_state[StateKeys.PLOT_SUBMITTED_CLICKED]
-):
-    st.session_state[StateKeys.PLOT_SUBMITTED_COUNTER] = st.session_state[
-        StateKeys.PLOT_SUBMITTED_CLICKED
-    ]
-    volcano_plot = gui_volcano_plot_differential_expression_analysis(
-        chosen_parameter_dict
-    )
-    volcano_plot._update(plotting_parameter_dict)
-    volcano_plot._annotate_result_df()
-    volcano_plot._plot()
-    genes_of_interest_colored = volcano_plot.get_colored_labels()
+    # TODO move this to volcano anyway ?
```

Comment:
These should ideally be functions of all analyses. To get a list of all significant proteins and be able to look them up on uniprot.